support/c/idris_file.c

```diff
@@ -61,7 +61,11 @@ int idris2_removeFile(const char *filename) {
 }
 
 int idris2_fileSize(FILE* f) {
+#ifdef _WIN32
+    int fd = win32_getFileNo(f);
+#else
     int fd = fileno(f);
+#endif
```

Comment:
Rather than repeating this code snippet multiple times, shouldn't we pull it out into
its own function?

Reply:
Yeah. I agree. One of those things that bugged me but not quite enough to fix as I wrote the code. I’ll abstract into a helper function that chooses between the two variants.

Reply:
Part of what stopped me initially is the thought that the whole structure of the windows specific functions begs for abstraction where there is currently duplication, but what I’ve done here is a particularly large amount of duplication. Good call out.